### PR TITLE
libssh: fix "empty expression statement has no effect" warnings

### LINKED ISSUE #6847: libssh: myards of warnings when compiling with libssh
### I did this
Compiling curl with `./configure CC=clang --enable-debug --with-libssh --prefix=$HOME/tmp`
```
vssh/libssh.c:706:47: warning: empty expression statement has no effect; remove unnecessary ';' to silence this warning [-Wextra-semi-stmt]
        MOVE_TO_ERROR_STATE(CURLE_FAILED_INIT);
                                              ^
vssh/libssh.c:716:60: warning: empty expression statement has no effect; remove unnecessary ';' to silence this warning [-Wextra-semi-stmt]
        MOVE_TO_ERROR_STATE(CURLE_PEER_FAILED_VERIFICATION);
                                                           ^
vssh/libssh.c:737:50: warning: empty expression statement has no effect; remove unnecessary ';' to silence this warning [-Wextra-semi-stmt]
          MOVE_TO_ERROR_STATE(CURLE_LOGIN_DENIED);
                                                 ^
vssh/libssh.c:755:50: warning: empty expression statement has no effect; remove unnecessary ';' to silence this warning [-Wextra-semi-stmt]
          MOVE_TO_ERROR_STATE(CURLE_LOGIN_DENIED);
                                                 ^
vssh/libssh.c:762:31: warning: empty expression statement has no effect; remove unnecessary ';' to silence this warning [-Wextra-semi-stmt]
        MOVE_TO_SECONDARY_AUTH;
...
```

### I expected the following
No warnings regarding unnecessary semicolons

### curl/libcurl version

curl: f573998c22a20b0fbe4edb8dbd968aa2f7ca5864
libssh: 0.9.5

### operating system
`FreeBSD freebsd 12.2-RELEASE-p4 FreeBSD 12.2-RELEASE-p4 GENERIC  amd64`

### PR DESCRIPTION
**Pull request by bagder**
... by fixing macros to do-while constructs and moving out the calls to
"break" outside of the actual macro. It also fixes the problem where the
macro was used witin a loop and the break didn't do right.

Reported-by: Emil Engler
Fixes #6847

### CODE REVIEW

**Comment by bagder** [comment]
@emilengler what do you think about this?

File: lib/vssh/libssh.c [hunk 1]
```diff
@@ -1292,8 +1302,11 @@ static CURLcode myssh_statemach_act(struct Curl_easy *data, bool *block)
                  CURL_READFUNC_ABORT return code still aborts */
               failf(data, "Failed to read data");
               MOVE_TO_ERROR_STATE(CURLE_FTP_COULDNT_USE_REST);
+              break;
             }
           } while(passed < data->state.resume_from);
+          if(rc)
```
Comment (emilengler) [inline lib/vssh/libssh.c:1308]: nit: Maybe do `rc != SSH_OK` (see line 776)

File: lib/vssh/libssh.c [hunk 2]
```diff
@@ -549,49 +549,48 @@ static int myssh_is_known(struct Curl_easy *data)
...
+    sshc->actualcode =                                          \
+      sftp_error_to_CURLE(sftp_get_error(sshc->sftp_session));  \
+    rc = SSH_ERROR;                                             \
+  } while(0)
 
-#define MOVE_TO_LAST_AUTH \
-  if(sshc->auth_methods & SSH_AUTH_METHOD_PASSWORD) { \
-    rc = SSH_OK; \
-    state(data, SSH_AUTH_PASS_INIT); \
-    break; \
```
Comment (emilengler) [inline lib/vssh/libssh.c:570]: I'm personally afraid of what happens to the break statements that depend on an if-logic and are not getting removed. Have you some deeper notes/researches about this "scenario"?

Comment (bagder) [reply lib/vssh/libssh.c:570]: They use `MOVE_TO_*` macros in the else statements too and those macros use `break` as well, so they're all doing break in both branches.

File: lib/vssh/libssh.c (hunk 1 above)
Comment (bagder) [reply lib/vssh/libssh.c:1308]: Right, but I prefer `if(rc)` than the other ways as `SSH_OK` is defined to zero.If you search in the libssh.c file you can also see a lot of `if(rc != 0)` checks, so it's not even consistently done already before.

**Pull request closed without merging**